extensions/io/org.eclipse.smarthome.io.mqttembeddedbroker/src/main/java/org/eclipse/smarthome/io/mqttembeddedbroker/internal/EmbeddedBrokerServiceImpl.java

```diff
@@ -0,0 +1,304 @@
+        initialize(new Configuration(data).as(ServiceConfiguration.class));
+    }
+
+    public void initialize(ServiceConfiguration config) {
+        if (config.port == null) {
+            config.port = config.secure ? 8883 : 1883;
+        }
+
+        // Create MqttBrokerConnection
+        connection = service.getBrokerConnection(Constants.CLIENTID);
+        if (connection != null) {
+            // Close the existing connection and remove it from the service
+            connection.stop();
+            service.removeBrokerConnection(Constants.CLIENTID);
+        }
+
+        connection = new MqttBrokerConnection("127.0.0.1", config.port, config.secure, Constants.CLIENTID);
+        connection.addConnectionObserver(this);
+
+        if (config.username != null) {
+            connection.setCredentials(config.username, config.password);
+        }
+
+        // Start embedded server
+        try {
+            startEmbeddedServer(config.port, config.secure, config.username, config.password, config.persistenceFile);
+        } catch (IOException e) {
+            logger.debug("Could not start embedded broker", e);
+            logger.debug(e.getLocalizedMessage());
+            return;
+        }
+    }
+
+    @Deactivate
+    public void deactivate() {
+        if (service != null) {
+            service.removeBrokersListener(this);
+        }
+        if (connection == null && server != null) {
+            server.stopServer();
+            server = null;
+            return;
+        }
+
+        // Clean shutdown: Stop connection, wait for process to finish, shutdown server
+        connection.removeConnectionObserver(this);
+        try {
+            connection.stop().thenRun(() -> {
+                if (server != null) {
+                    server.stopServer();
+                    server = null;
+                }
+            }).get(300, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException | ExecutionException | TimeoutException ignored) {
+        }
+        connection = null;
+    }
+
+    @Override
+    public void brokerAdded(String brokerID, MqttBrokerConnection broker) {
+    }
+
+    @Override
+    public void brokerRemoved(String brokerID, MqttBrokerConnection broker) {
+        // Do not allow this connection to be removed. Add it again.
+        if (broker == connection) {
+            service.addBrokerConnection(brokerID, broker);
+        }
+    }
+
+    /**
+     * Starts the embedded broker.
+     *
+     * @param port The broker port.
+     * @param secure Allow only secure connections if true or only plain connections otherwise.
+     * @param username Broker authentication user name. May be null.
+     * @param password Broker authentication password. May be null.
+     * @param persistence_filename The filename were persistent data should be stored.
+     * @throws IOException If any error happens, like the port is already in use, this exception is thrown.
+     */
+    @Override
```

Comment:
If a method of an interface is implemented we already inherit the JavaDoc of the interface method.
Overwriting that comment makes only sense if you need to add additional information.
As a consumer of the interface implementation WRT to OSGi services does not know about the specific implementation it should be removed completely as the behavior should be exactly the ones described by the interface.